legal-api/tests/unit/api/test_business_incorporation.py

```diff
@@ -66,51 +75,54 @@ def test_post_new_incorporation(session, client, jwt):
 
-    assert rv.status_code == HTTPStatus.CREATED
-    # Attempt a POST with the same NR
-    rv = client.post(f'/api/v1/businesses',
-                     json=filing,
-                     headers=create_header(jwt, [STAFF_ROLE], nr_number))
+        assert rv.status_code == HTTPStatus.CREATED
+        # Attempt a POST with the same NR
+        rv = client.post(f'/api/v1/businesses',
+                         json=filing,
+                         headers=create_header(jwt, [STAFF_ROLE], nr_number))
 
-    assert HTTPStatus.BAD_REQUEST == rv.status_code
-    assert 'Incorporation filing for NR 1234567 already exists' == rv.json['errors'][0]['message']
+        assert HTTPStatus.BAD_REQUEST == rv.status_code
+        assert 'Incorporation filing for NR 1234567 already exists' == rv.json['errors'][0]['message']
 
 
 def test_get_incorporation_filing(session, client, jwt):
     """Assert that an incorporation filing can be retrieved for resuming."""
-    nr_number = 'NR 1234567'
     filing = copy.deepcopy(INCORPORATION_FILING_TEMPLATE)
     filing['filing']['incorporationApplication']['nameRequest']['nrNumber'] = nr_number
+    filing['filing']['header']['effectiveDate'] = effective_date
 
-    # Post initial filing
-    rv = client.post(f'/api/v1/businesses?draft=true',
-                     json=filing,
-                     headers=create_header(jwt, [STAFF_ROLE], nr_number))
+    # perform test
+    with freeze_time(now):
+        # Post initial filing
+        rv = client.post(f'/api/v1/businesses?draft=true',
+                         json=filing,
+                         headers=create_header(jwt, [STAFF_ROLE], nr_number))
 
-    assert HTTPStatus.CREATED == rv.status_code
-    assert rv.json['filing']['header']['filingId']
+        assert HTTPStatus.CREATED == rv.status_code
+        assert rv.json['filing']['header']['filingId']
 
-    filing_id = rv.json['filing']['header']['filingId']
+        filing_id = rv.json['filing']['header']['filingId']
 
-    # Retrieve the incorporation filing
-    rv = client.get(f'/api/v1/businesses/{nr_number}/filings/{filing_id}',
-                    headers=create_header(jwt, [STAFF_ROLE], nr_number))
+        # Retrieve the incorporation filing
+        rv = client.get(f'/api/v1/businesses/{nr_number}/filings/{filing_id}',
+                        headers=create_header(jwt, [STAFF_ROLE], nr_number))
 
-    assert HTTPStatus.OK == rv.status_code
-    assert filing_id == rv.json['filing']['header']['filingId']
+        assert HTTPStatus.OK == rv.status_code
+        assert filing_id == rv.json['filing']['header']['filingId']
 
 
 def test_put_draft_incorporation_filing(session, client, jwt):
     """Assert that an incorporation filing can be put (updated) to filings endpoint."""
-    nr_number = 'NR 1234567'
     filing_json = copy.deepcopy(INCORPORATION_FILING_TEMPLATE)
     filing_json['filing']['incorporationApplication']['nameRequest']['nrNumber'] = nr_number
```

Comment:
How is nr_number getting assigned now? It looks like this should cause an error to me, am I missing something?

Reply:
I defined `nr_number` as the `indentifier` at the beginning of the test setup, and assigned that identifier to the filing nrNumber in each test. Basically just defined the NR once opposed to defining it in each test. 

see line 31 for the definition

an example of how i assigned it in each test.
`f['filing']['incorporationApplication']['nameRequest']['nrNumber'] = identifier`

Reply:
Oh cool! Nice